esp-hal/src/gpio/mod.rs

```diff
@@ -1390,7 +1415,7 @@ macro_rules! gpio {
             }
 
             impl ErasedPin {
-                pub(crate) unsafe fn clone_unchecked(&mut self) ->  Self {
+                pub(crate) fn clone(&self) -> ErasedPin {
```

Comment:
I see why this is changed to take `&self` and given it's `pub(crate)` it doesn't matter much but shouldn't we keep `unsafe`

Reply:
Oh haha this went in a big circle. I'll undo the rename and re-add the unsafe